cairo/tests/src/utils/test_signature.py

```diff
@@ -1,32 +1,145 @@
-        eth_address=felt,
-    )
-    def test_should_raise_with_invalid_y_parity(
-        self, cairo_run, msg_hash: U256, r: U256, s: U256, y_parity, eth_address
-    ):
-        assume(r != 0 and s != 0)
-        assume(r < SECP256K1N and s < SECP256K1N)
-        with cairo_error("Invalid y_parity"):
+    class TestPublicKeyPointToEthAddress:
+        @given(private_key=...)
+        def test__public_key_point_to_eth_address(
+            self, cairo_run, private_key: PrivateKey
+        ):
+            public_key = private_key.public_key
+            public_key_bytes = public_key.to_bytes()
+            x = U256.from_be_bytes(public_key_bytes[0:32])
+            y = U256.from_be_bytes(public_key_bytes[32:64])
+
+            expected_address = public_key.to_address()
+            result = cairo_run(
+                "test__public_key_point_to_eth_address",
+                x=x,
+                y=y,
+            )
+            assert result == int(expected_address, 16)
+
+    class TestVerifyEthSignature:
+        @given(private_key=..., message=...)
+        @pytest.mark.slow
+        def test__verify_eth_signature_uint256(
+            self, cairo_run, private_key: PrivateKey, message: Bytes32
+        ):
+            signature = private_key.sign_msg_hash(message)
+            eth_address = int(private_key.public_key.to_address(), 16)
             cairo_run(
                 "test__verify_eth_signature_uint256",
-                msg_hash,
-                r,
-                s,
-                y_parity,
-                eth_address,
+                msg_hash=U256.from_be_bytes(message),
+                r=U256(signature.r),
+                s=U256(signature.s),
+                y_parity=signature.v,
+                eth_address=eth_address,
+            )
+
+        @given(
+            msg_hash=...,
+            r=st.one_of(st.integers(min_value=1, max_value=SECP256K1N - 1).map(U256)),
+            s=st.one_of(st.integers(min_value=1, max_value=SECP256K1N - 1).map(U256)),
+            y_parity=st.integers(min_value=2, max_value=DEFAULT_PRIME - 1),
+            eth_address=felt,
+        )
+        def test_should_raise_with_invalid_y_parity(
+            self, cairo_run, msg_hash: U256, r: U256, s: U256, y_parity, eth_address
+        ):
+            with cairo_error("Invalid y_parity"):
+                cairo_run(
+                    "test__verify_eth_signature_uint256",
+                    msg_hash,
+                    r,
+                    s,
+                    y_parity,
+                    eth_address,
+                )
+
+        @given(
+            msg_hash=...,
+            r=st.one_of(
+                st.just(U256(0)),
+                st.integers(min_value=SECP256K1N, max_value=2**256 - 1).map(U256),
+            ),
+            s=...,
+            y_parity=...,
+            eth_address=felt,
+        )
+        @settings(max_examples=5)
+        def test_should_raise_with_out_of_bounds_r(
```

Comment:
```suggestion
        @settings(max_examples=5)
        @given(
            msg_hash=...,
            r=st.one_of(
                st.just(U256(0)),
                st.integers(min_value=SECP256K1N, max_value=2**256 - 1).map(U256),
            ),
            s=...,
            y_parity=...,
            eth_address=felt,
        )
        def test_should_raise_with_out_of_bounds_r(
```
I would suggest to put settings and related conf test before parameters to avoid break the flow of thoughts